### PR TITLE
Add configurable announcement for course instructors

### PR DESCRIPTION
**Pull request by nwalters512**
This will be used in an upcoming migration of some courses from `ca.prairielearn.com` to `us.prairielearn.com`. This will allow us to warn instructors and students that they should access the course from the new domain.

Screenshot 2024-06-04 at 13 30 42

These settings mirror the global announcement banner:

https://github.com/PrairieLearn/PrairieLearn/blob/c2d07852aef77dd89214f9164435ba535f9c807e/apps/prairielearn/src/pages/partials/navbar.ejs#L12-L16

https://github.com/PrairieLearn/PrairieLearn/blob/c2d07852aef77dd89214f9164435ba535f9c807e/apps/prairielearn/src/lib/config.ts#L431-L440

### CODE REVIEW

File: apps/prairielearn/src/lib/db-types.ts [hunk 1]
```diff
@@ -88,6 +88,8 @@ export const ModeSchema = z.enum(['Public', 'Exam', 'SEB']);
 export type Mode = z.infer<typeof ModeSchema>;
 
 export const CourseSchema = z.object({
+  banner_color: z.string().nullable(),
+  banner_html: z.string().nullable(),
```
Comment (nwalters512) [inline apps/prairielearn/src/lib/db-types.ts:92]: We could name these `announcement_*` for consistency with the global one?

File: apps/prairielearn/src/pages/partials/navbar.ejs [hunk 2]
```diff
@@ -345,6 +345,12 @@
   </div>
 </nav>
 
+<% if (typeof course !== 'undefined' && course.banner_html && course.banner_color) { %>
+  <div class="alert alert-<%= course.banner_color %> mb-0 rounded-0 text-center">
+    <%- course.banner_html %>
+  </div>
+<% } %>
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/partials/navbar.ejs:352]: This currently shows on all course pages. We could in theory use the value of `navPage` to limit this to instructor-facing pages?

File: database/tables/pl_courses.pg [hunk 3]
```diff
@@ -1,4 +1,6 @@
 columns
+    banner_color: text
+    banner_html: text
```
Comment (nwalters512) [inline database/tables/pl_courses.pg:3]: I didn't build editing for this into the admin UI; the expectation is that we'll set this in bulk with a manual query.

Comment (mwest1066) [reply database/tables/pl_courses.pg:3]: Very reasonable.

File: apps/prairielearn/src/lib/db-types.ts (hunk 1 above)
Comment (mwest1066) [reply apps/prairielearn/src/lib/db-types.ts:355]: I'm (weakly) in favor of that.

File: apps/prairielearn/src/pages/partials/navbar.ejs (hunk 2 above)
Comment (mwest1066) [reply apps/prairielearn/src/pages/partials/navbar.ejs:353]: That's probably a good idea. That way instructors can continue to use the existing CA course instances to run makeup exams at the start of the new semester, without confusing students with a "this course has moved" messages.

**Review by mwest1066 — APPROVED** [review]

File: apps/prairielearn/src/pages/partials/navbar.ejs (hunk 2 above)
Comment (nwalters512) [reply apps/prairielearn/src/pages/partials/navbar.ejs]: Done in da38600.

File: apps/prairielearn/src/lib/db-types.ts (hunk 1 above)
Comment (nwalters512) [reply apps/prairielearn/src/lib/db-types.ts]: Changed in da38600.

**Review by mwest1066 — APPROVED** [review]

**Pull request merged**